### PR TITLE
For those who use no composer, a local autoload is good.

### PR DESCRIPTION
**Pull request by paulorwd**
For those who do not use the composer, un autoload own.

### CODE REVIEW

File: autoload.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+//
+//     require_once "<path-to-here>/Onedrive/autoload.php"
+
+/**
+ * @internal
+ */
+function autoload($name)
+{
+    // If the name doesn't start with "Krizalys\Onedrive\", then its not once of our classes.
+    if (\substr_compare($name, "Krizalys\\Onedrive\\", 0, 19) !== 0) return;
```
Comment (krizalys) [inline autoload.php]: `18` no?

File: autoload.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    // Take the "Krizalys\\Onedrive\" prefix off.
+    $stem = \substr($name, 19);
```
Comment (krizalys) [inline autoload.php]: `18` no?

File: autoload.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<?php
```
Comment (krizalys) [inline autoload.php]: For consistency among other PHP files, there should be an empty line here.

File: autoload.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+// But if you really don't want to use a global autoloader, do this:
+//
+//     require_once "<path-to-here>/Onedrive/autoload.php"
+
+/**
+ * @internal
+ */
+function autoload($name)
+{
+    // If the name doesn't start with "Krizalys\Onedrive\", then its not once of our classes.
```
Comment (krizalys) [inline autoload.php]: Typo `once` => `one`.

File: autoload.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    // Convert "\" and "_" to path separators.
+    $pathified_stem = \str_replace(array("\\", "_"), '/', $stem);
+
+    $path = __DIR__ . "/src/Krizalys/Onedrive/" . $pathified_stem . ".php";
+    if (\is_file($path)) {
+        require_once $path;
+    }
+}
+
+\spl_autoload_register('Onedrive\autoload');
```
Comment (krizalys) [inline autoload.php]: There should also be a trailing newline here.

File: autoload.php [hunk 6]
```diff
@@ -0,0 +1,31 @@
+<?php
+namespace Onedrive;
```
Comment (krizalys) [inline autoload.php]: Let's please be consistent and use the same `namespace Krizalys\Onedrive;` everywhere.

**Comment by krizalys** [comment]
Thanks for this. I'll probably not use it personally, but I'm OK to merge it if you address my comments.

2 other things which matter to me but would have overloaded your PR :)

* There is no need to use `\` in front of global functions, I find it too verbose, difficult to type, and not consistent with the rest of the PHP files
* Variable names generally use `camelCase` in this project

File: autoload.php (hunk 1 above)
Comment (paulorwd) [inline autoload.php]: 19 thereof, includes a '/'.

File: autoload.php (hunk 3 above)
Comment (paulorwd) [inline autoload.php]: Ok

**Comment by paulorwd** [comment]
Yes, I will make the necessary changes yes, thank you.

File: autoload.php [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+// The Onedrive SDK autoloader.  You probably shouldn't be using this.  Instead,
+// use a global autoloader, like the Composer autoloader.
+//
+// But if you really don't want to use a global autoloader, do this:
+//
+//     require_once "<path-to-here>/Onedrive/autoload.php"
```
Comment (krizalys) [inline autoload.php]: This path currently seems incorrect, it should be `<path-to-here>/autoload.php`, but we should probably just say `__DIR__ . '/autoload.php'` as the `<path-to-here>` is confusing (path to this file? or path to the containing directory?).

**Comment by paulorwd** [comment]
All of us as?

File: autoload.php [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    // Convert "\" and "_" to path separators.
+    $pathifiedStem = str_replace(array('\\', '_'), '/', $stem);
+
+    $path = __DIR__ . '/src/Krizalys/Onedrive/' . $pathifiedStem . '.php';
+    if (is_file($path)) {
+        require_once $path;
+    }
+}
+
+spl_autoload_register('Onedrive\autoload');
```
Comment (krizalys) [inline autoload.php]: Needs to be adjusted to match the namespace: `Krizalys\Onedrive`.

File: autoload.php (hunk 2 above)
Comment (krizalys) [inline autoload.php]: I still believe that it should be `18`:

```php
echo substr('Krizalys\Onedrive\Client', 19);
```

=> `lient`

File: autoload.php [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+/**
+ * @internal
+ */
+function autoload($name)
+{
+    // If the name doesn't start with "Krizalys\Onedrive\", then it's not one of our classes.
+    if (substr_compare($name, 'Krizalys\\Onedrive\\', 0, 19) !== 0) return;
+
+    // Take the "Krizalys\\Onedrive\" prefix off.
+    $stem = substr($name, 19);
```
Comment (krizalys) [inline autoload.php]: I still believe that it should be 18:

echo substr('Krizalys\Onedrive\Client', 19);
=> lient

File: autoload.php [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+//
+//     require_once __DIR__ . '/Onedrive/autoload.php'
+
+/**
+ * @internal
+ */
+function autoload($name)
+{
+    // If the name doesn't start with "Krizalys\Onedrive\", then it's not one of our classes.
+    if (substr_compare($name, 'Krizalys\\Onedrive\\', 0, 19) !== 0) return;
```
Comment (krizalys) [inline autoload.php]: Same as below.

File: autoload.php (hunk 9 above)
Comment (paulorwd) [inline autoload.php]: In tests I conducted, the name of the class is that way "Krizalys\\Onedrive\\Client", yes, with two bars, so I used 19.

File: autoload.php (hunk 8 above)
Comment (paulorwd) [inline autoload.php]: Fixed.

File: autoload.php (hunk 9 above)
Comment (paulorwd) [inline autoload.php]: Really, corrected.

**Pull request merged**